crates/rustc_codegen_spirv/src/builder/intrinsics.rs

```diff
@@ -398,6 +367,254 @@ impl<'a, 'tcx> IntrinsicCallBuilderMethods<'tcx> for Builder<'a, 'tcx> {
 }
 
 impl Builder<'_, '_> {
+    pub fn count_ones(&self, arg: SpirvValue) -> SpirvValue {
+        let ty = arg.ty;
+        match self.cx.lookup_type(ty) {
+            SpirvType::Integer(bits, signed) => {
+                let u32 = SpirvType::Integer(32, false).def(self.span(), self);
+
+                match bits {
```

Comment:
Just a style thing, but I generally feel it is clearer to matched on `signed` as well, it makes all the cases clearer, similar to what you did in bitcast.